### PR TITLE
KRKNWK-7497: zigbee: Enable MPSL in all Zigbee examples

### PR DESCRIPTION
**Pull request by tomchy**
Change configuration of Zigbee examples to use MPSL in both single and multiprotocol examples.

Signed-off-by: Tomasz Chyrowicz <tomasz.chyrowicz@nordicsemi.no>

### CODE REVIEW

**Comment by eugmes** [comment]
There is a typo in the commit message "single na dmultiprotocol".

File: samples/zigbee/light_bulb/prj.conf [hunk 1]
```diff
@@ -43,3 +39,11 @@ CONFIG_NET_IPV6_NBR_CACHE=n
 CONFIG_NET_IPV6_RA_RDNSS=n
 CONFIG_NET_IP_ADDR_CHECK=n
 CONFIG_NET_UDP=n
+
+# BLE stack is required for MPSL compatible flash driver.
+CONFIG_BT=y
+CONFIG_BT_HCI=y
+CONFIG_BT_CTLR=y
+CONFIG_BT_LL_NRFXLIB=y
```
Comment (edmont) [inline samples/zigbee/light_bulb/prj.conf:47]: This one and the following are undefined symbols. Maybe something has changed with the upmerge?

Comment (tomchy) [reply samples/zigbee/light_bulb/prj.conf:47]: Yes, they are undefined. This PR is not ready for a review. This change is introduced by a simple cherry-pick of a commit from the past.

Comment (tomchy) [reply samples/zigbee/light_bulb/prj.conf]: Building fixed!
I tried to go through Kconfig diffs and disable some of the BLE features. I will keep them removed if they have any impact on the flash & RAM usage.

File: samples/zigbee/light_bulb/prj.conf [hunk 2]
```diff
@@ -43,3 +39,7 @@ CONFIG_NET_IPV6_NBR_CACHE=n
 CONFIG_NET_IPV6_RA_RDNSS=n
 CONFIG_NET_IP_ADDR_CHECK=n
 CONFIG_NET_UDP=n
+
+# BLE stack is required for MPSL compatible flash driver.
```
Comment (rlubos) [inline samples/zigbee/light_bulb/prj.conf:43]: This is still not fixed? 😕 This special flash driver shall be enabled by MPSL if it's required by it, please push MPSL guys to sort it out.

**Review by rlubos — APPROVED** [review]
LGTM

File: samples/zigbee/light_bulb/prj.conf (hunk 2 above)
Comment (eugmes) [reply samples/zigbee/light_bulb/prj.conf:43]: @rlubos The MPSL compatible flash driver PR is here: https://github.com/nrfconnect/sdk-nrf/pull/3067

Comment (edmont) [reply samples/zigbee/light_bulb/prj.conf:43]: Enabling BT fills all the flash size for some OT samples. We are hoping this gets merged on time: https://github.com/nrfconnect/sdk-nrf/pull/3067

Comment (rlubos) [reply samples/zigbee/light_bulb/prj.conf:43]: @eugmes Thanks for the info, great!

@tomchy Perhaps we should postpone this PR until #3067 gets merged, I see it's targeting 1.4.

Comment (tomchy) [reply samples/zigbee/light_bulb/prj.conf:43]: They work on it right now ;)
https://github.com/nrfconnect/sdk-nrf/pull/3067

Comment (tomchy) [reply samples/zigbee/light_bulb/prj.conf:43]: Well - if we wait for this PR, than the second commit has to be removed.

Comment (tomchy) [reply samples/zigbee/light_bulb/prj.conf]: The workaround removed from the PR. Now the #3067 has to be merged prior this PR.

**Comment by rlubos** [comment]
https://github.com/nrfconnect/sdk-nrf/pull/3067 got merged

**Comment by tomchy** [comment]
PR rebased. CI restarted.

**Review by rlubos — APPROVED** [review]

**Comment by tomchy** [comment]
I see some CI failures. Still waiting for all jobs to investigate through artifacts.

**Comment by tomchy** [comment]
Functionally - it works. For some reason - the MCUboot refused to perform firmware upgrade. I have a feeling that this issue is not related to this PR. Test restarted to verify if this issue can be easily reproduced.

**Comment by tomchy** [comment]
Issue reproduced for the second time. Running tests on the master branch to see if the regression is introduced by this PR.

**Comment by tomchy** [comment]
Issue reproduced on the master branch. PR ready to be merged!

**Review by carlescufi — APPROVED** [review]

**Pull request merged**